src/jit/lsra.cpp

```diff
@@ -5680,16 +5679,14 @@ regNumber LinearScan::tryAllocateFreeReg(Interval* currentInterval, RefPosition*
 
     if (availablePhysRegInterval != nullptr)
     {
-        if (intervalToUnassign != nullptr)
+        intervalToUnassign = availablePhysRegInterval->assignedInterval;
+        if (isAssigned(availablePhysRegInterval ARM_ARG(currentInterval->registerType)))
         {
-            RegRecord* physRegToUnassign = availablePhysRegInterval;
 #ifdef _TARGET_ARM_
-            // We should unassign a double register if availablePhysRegInterval is part of the double register
-            if (availablePhysRegInterval->assignedInterval->registerType == TYP_DOUBLE &&
-                !genIsValidDoubleReg(availablePhysRegInterval->regNum))
-                physRegToUnassign = findAnotherHalfRegRec(availablePhysRegInterval);
+            unassignPhysReg(availablePhysRegInterval, currentInterval->registerType);
+#else
+            unassignPhysReg(availablePhysRegInterval, intervalToUnassign->recentRefPosition);
```

Comment:
@CarolEidt OK. I've changed to merge this codes into one code for all architectures like what you said. Thank you.